batch/test/test_batch.py

```diff
@@ -1128,8 +1121,88 @@ def test_update_batch_w_deps_in_update(client: BatchClient):
     status = j2.wait()
 
     assert status['state'] == 'Success', str((status, b.debug_info()))
-    assert j1.job_id == 2, str(b.debug_info())
-    assert j2.job_id == 3, str(b.debug_info())
+    assert b.status()['state'] == 'success', str((b.status(), b.debug_info()))
+    assert b.status()['n_jobs'] == 3, str(b.debug_info())
+
+
+def test_update_batch_w_deps_in_update_always_run(client: BatchClient):
+    bb = client.create_batch()
+    j = bb.create_job(DOCKER_ROOT_IMAGE, ['false'])
+    bb.submit()
+
+    j.wait()
+
+    j1 = bb.create_job(DOCKER_ROOT_IMAGE, ['true'], always_run=True)
+    j2 = bb.create_job(DOCKER_ROOT_IMAGE, ['true'], parents=[j, j1])
+    b = bb.submit()
+    status = j2.wait()
+
+    assert status['state'] == 'Success', str((status, b.debug_info()))
```

Comment:
Shouldn't `j1.wait()['state'] == 'Success'` and `j2.wait()['state'] == 'Cancelled'`? Can we group the call to wait and the assertion together so it's clear which job state we're looking at? I like how the test below this one is organized